sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/multi/MultiServiceEventClient.java

```diff
@@ -446,15 +456,22 @@ public UnexpectedResourcesResponse getUnexpectedResources(Collection<Protos.Offe
      */
     @Override
     public TaskStatusResponse taskStatus(Protos.TaskStatus status) {
-        Optional<AbstractScheduler> service = multiServiceManager.getMatchingService(status);
-        if (!service.isPresent()) {
-            // Unrecognized service. Status for old task?
-            LOGGER.info("Received status for unknown task {}: {}",
-                    status.getTaskId().getValue(), TextFormat.shortDebugString(status));
-            return TaskStatusResponse.unknownTask();
-        }
-        LOGGER.info("Received status for task {}: {}", status.getTaskId().getValue(), status.getState());
-        return service.get().taskStatus(status);
+        return multiServiceManager
+                .getMatchingService(status)
+                .map(x -> x.taskStatus(status))
```

Comment:
The first call to `.map(x -> x.taskStatus(status))` is for mapping the matched `AbstractScheduler` to `TaskStatusResponse`. If a matching service was found the second `.map(x -> x.taskStatus(status))` would never be executed.

`.map(x -> x.taskStatus(status))` is just converting `Optional<AbstractScheduler>` to `Optional<TaskStatusResponse>`

If a matching service was not found, we assume that the task status was meant for default service.

`orElseGet` would only be executed when the `getMatchingService` returns and `empty` optional.

This is whats happening in the call. 
```
                .getMatchingService(status) // See if a matching service is found and return an Optional
                .map(x -> x.taskStatus(status)) // If a matching `AbstractScheduler` is found, map it to a `TaskStatusResponse` object. This would happen only if above optional is non empty.
                .orElseGet(() -> multiServiceManager // only enter this block if getMatchingService returned an empty optional.
                    ....
```